### PR TITLE
windows dev docs - vagrant info

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
I've finally got some Vagrant boxes built and ready to be used for Ansible module development. This PR adds in info on how to use them and some background info.

Also added in a module util that was recently merged in.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
windows dev guide

##### ANSIBLE VERSION
```
2.5
```

### CODE REVIEW

**Comment by dharmabumstead** [comment]
Doc review/edit in process...please do not merge.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst [hunk 1]
```diff
@@ -15,8 +15,145 @@ What's covered in this section:
...
+
+.. note:: The port ``55986`` is automatically forwarded by Vagrant to the
+    Windows host that was created, if this conflicts with an existing local
+    port then Vagrant will automatically use another one at random and display
+    show that in the output.
+
+The OS that is created is based on the image set. The following
+images can be used:
+
+- `jborean93/WindowsServer2008-x86 <https://app.vagrantup.com/jborean93/boxes/WindowsServer2008-x86>`_
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules_general_windows.rst:83]: Please rename the images so they don't feature your username.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst [hunk 2]
```diff
@@ -15,8 +15,145 @@ What's covered in this section:
...
+- `jborean93/WindowsServer2012R2 <https://app.vagrantup.com/jborean93/boxes/WindowsServer2012R2>`_
+- `jborean93/WindowsServer2016 <https://app.vagrantup.com/jborean93/boxes/WindowsServer2016>`_
+
+When the host is online, it can accessible by RDP on ``127.0.0.1:3389`` but the
+port may differ depending if there was a conflict. To get rid of the host, run
+``vagrant destroy --force`` and Vagrant will automatically remove the VM and
+any other files associated with that VM.
+
+While this is useful when testing modules on a single Windows instance, these
+hosts won't work out of the box with domain based modules. The Vagrantfile at
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules_general_windows.rst]: Please avoid colloquialisms like "out of the box". Suggest revising to something easier to localize, like "won't work without modification".

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Made some edits. Need to think about the organization - try to outline it, then use the outline items as sub headers to make it easier to follow.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst (hunk 1 above)
Comment (jborean93) [reply docs/docsite/rst/dev_guide/developing_modules_general_windows.rst:83]: I can't, these images are not supported/managed by Ansible and are just registered under my name. Vagrant Cloud always puts the username as a prefix on the boxes.

I don't think we want to manage these under an Ansible umbrella and it was something I've been doing in my spare time.

File: docs/docsite/rst/dev_guide/developing_modules_general_windows.rst (hunk 2 above)
Comment (jborean93) [reply docs/docsite/rst/dev_guide/developing_modules_general_windows.rst]: Will do, thanks.

**Comment by jborean93** [comment]
@dharmabumstead I've updated some of things you've requested and added some comments on things I can't really do. Let me know your thoughts on those.

ready_for_review

**Review by dharmabumstead — APPROVED** [review]

**Comment by dharmabumstead** [comment]
Thanks @jborean93

**Pull request merged**